apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/ErrorReportingSnippets.cs

```diff
@@ -0,0 +1,193 @@
+using Google.Cloud.Diagnostics.Common.Tests;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Snippets
+{
+    [SnippetOutputCollector]
+    public class ErrorReportingSnippetsTests : IDisposable
+    {
+        private static readonly bool s_isWindows = Utils.IsWindows();
+        private static readonly ErrorEventEntryPolling s_polling = new ErrorEventEntryPolling();
+
+        private readonly string _testId;
+
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        private readonly DateTime _startTime;
+
+        public ErrorReportingSnippetsTests()
+        {
+            _testId = Utils.GetTestId();
+
+            var builder = new WebHostBuilder().UseStartup<ErrorReportingTestApplication>();
+            _server = new TestServer(builder);
+            _client = _server.CreateClient();
+
+            _startTime = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Test for the snippet sample ReportUnhandledExceptions which is a
+        /// configuration sample. Only way to test that the configuration is ok
+        /// is to check that logging of errors works as expected.
+        /// </summary>
+        [Fact]
+        public async Task Logs_UnhandledException()
+        {
+            await Assert.ThrowsAsync<Exception>(() => _client.GetAsync($"/ErrorLoggingSamples/ThrowsException/{_testId}"));
+
+            PollAndVerifyErrorEvent(_startTime, _testId, "ThrowsException");
+        }
+
+        /// <summary>
+        /// Test for the snippet sample LogExceptions. On the snippet we need to show 
+        /// the method signature to show injection of the logger from services, and also
+        /// we need the method to live in a MVC controller so that the logger is properly
+        /// injected. Because of that we can't have the snippet being the test itself.
+        /// </summary>
+        [Fact]
+        public async Task Logs_Explicitly()
+        {
+            await _client.GetAsync($"/ErrorLoggingSamples/CatchesAndLogsException/{_testId}");
+
+
+            // Verifying with function name ThrowsExceptions because that is the function
+            // that actually throws the Exception so will be the one included as
+            // ErrorEvent.Context.ReportLocation.FunctionName
+            PollAndVerifyErrorEvent(_startTime, _testId, "ThrowsException");
+        }
+
+        internal static void PollAndVerifyErrorEvent(DateTime startTime, string testId, string functionName)
+        {
+            var errorEvents = s_polling.GetEvents(startTime, testId, 1);
+            Assert.Single(errorEvents);
+            var errorEvent = errorEvents.First();
```

Comment:
You can just do errorEvents.Single() and combine these two lines.

Reply:
Done.